dart/lib/src/client.dart

```diff
@@ -188,48 +86,69 @@ abstract class SentryClient {
     options.httpClient?.close();
   }
 
-  @override
-  String toString() => '$SentryClient("$postUri")';
-
-  @protected
-  List<int> bodyEncoder(Map<String, dynamic> data, Map<String, String> headers);
-
-  @protected
-  @mustCallSuper
-  Map<String, String> buildHeaders(String authHeader) {
-    final headers = {
-      'Content-Type': 'application/json',
-    };
-
-    if (authHeader != null) {
-      headers['X-Sentry-Auth'] = authHeader;
+  SentryEvent _processEvent(
+    SentryEvent event, {
+    dynamic hint,
+    List<EventProcessor> eventProcessors,
+  }) {
+    for (final processor in eventProcessors) {
+      try {
+        event = processor(event, hint);
+      } catch (err) {
+        options.logger(
+          SentryLevel.error,
+          'An exception occurred while processing event by a processor : $err',
+        );
+      }
+      if (event == null) {
+        options.logger(SentryLevel.debug, 'Event was dropped by a processor');
+        break;
+      }
     }
-
-    return headers;
+    return event;
```

Comment:
still missing the execution of the beforeSendCallback ->
https://github.com/getsentry/sentry-java/blob/5ebd49cbfd17eb66175e9247db6d27e4df84a56b/sentry/src/main/java/io/sentry/SentryClient.java#L97-L103